src/dpp/application.cpp

```diff
@@ -100,28 +86,85 @@ application& application::fill_from_json_impl(nlohmann::json* j) {
 			this->team.members.emplace_back(tm);
 		}
 	}
+
+	set_snowflake_not_null(j, "guild_id", guild_id);
+	guild_obj = guild().fill_from_json(&((*j)["guild"]));
+	set_snowflake_not_null(j, "primary_sku_id", primary_sku_id);
+	set_string_not_null(j, "slug", slug);
+	set_iconhash_not_null(j, "cover_image", cover_image);
+	set_int32_not_null(j, "flags", flags);
+	set_int64_not_null(j, "approximate_guild_count", approximate_guild_count);
+
+	if (j->contains("redirect_uris")) {
+		for (const auto& uri : (*j)["redirect_uris"]) {
+			this->redirect_uris.push_back(to_string(uri));
+		}
+	}
+
+	set_string_not_null(j, "interactions_endpoint_url", interactions_endpoint_url);
 	set_string_not_null(j, "role_connections_verification_url", role_connections_verification_url);
+
+	if (j->contains("tags")) {
+		for (const auto& tag : (*j)["tags"]) {
+			this->tags.push_back(to_string(tag));
+		}
+	}
+
+	if (j->contains("install_params")) {
+		json& p = (*j)["install_params"];
+		set_snowflake_not_null(&p, "permissions", this->install_params.permissions);
+		for (const auto& scope : p["scopes"]) {
+			this->install_params.scopes.push_back(to_string(scope));
+		}
+	}
+
+	set_string_not_null(j, "custom_install_url", custom_install_url);
+
+	// TODO: Investigate https://discord.com/developers/docs/resources/application#application-resource when v11 releases. See if the variables below are documented.
+
+	set_int8_not_null(j, "discoverability_state", discoverability_state);
+	set_int32_not_null(j, "discovery_eligibility_flags", discovery_eligibility_flags);
+	set_int8_not_null(j, "explicit_content_filter", explicit_content_filter);
+	set_int8_not_null(j, "creator_monetization_state", creator_monetization_state);
+	set_bool_not_null(j, "integration_public", integration_public);
+	set_bool_not_null(j, "integration_require_code_grant", integration_require_code_grant);
+
+	if (j->contains("interactions_event_types")) {
+		for (const auto& event_type : (*j)["interactions_event_types"]) {
+			this->interactions_event_types.push_back(to_string(event_type));
+		}
+	}
+
+	set_int8_not_null(j, "interactions_version", interactions_version);
+	set_bool_not_null(j, "is_monetized", is_monetized);
+	set_int32_not_null(j, "monetization_eligibility_flags", monetization_eligibility_flags);
+	set_int8_not_null(j, "monetization_state", monetization_state);
+	set_bool_not_null(j, "hook", hook);
+	set_int8_not_null(j, "rpc_application_state", rpc_application_state);
+	set_int8_not_null(j, "store_application_state", store_application_state);
+	set_int8_not_null(j, "verification_state", verification_state);
+
 	return *this;
 }
 
 std::string application::get_cover_image_url(uint16_t size, const image_type format) const {
-	if (!this->cover_image.to_string().empty() && this->id) {
+	if (!this->cover_image.to_string().empty()) {
```

Comment:
the id is required here, as it is used in the url, else youll have app-icons/0/xxxxxx